src/ucp/wireup/select.c

```diff
@@ -116,7 +116,7 @@ static int ucp_wireup_is_reachable(ucp_worker_h worker, ucp_rsc_index_t rsc_inde
 {
     ucp_context_h context = worker->context;
     return (context->tl_rscs[rsc_index].tl_name_csum == ae->tl_name_csum) &&
-           uct_iface_is_reachable(worker->ifaces[rsc_index], ae->dev_addr);
+           uct_iface_is_reachable(worker->ifaces[rsc_index], ae->dev_addr, ae->iface_addr);
```

Comment:
@yosefe `dev_addr` and `iface_addr` are pointers to the addresses structure. They always points to the memory as I described in [878](https://github.com/openucx/ucx/issues/878) and contains trash if add_len == 0. I think we have to set them to NULL explicitly here or in address_unpuck procedure. The second one requires memory allocation for the addresses.